code/controllers/subsystem/autofire.dm

```diff
@@ -210,7 +210,7 @@ SUBSYSTEM_DEF(automatedfire)
 	shooting = FALSE
 
 /datum/component/automatedfire/xeno_turret_autofire/process_shot()
-	if(!shooting)
+	if(!shooting || !parent)
 		return
```

Comment:
how would there be no parent

Reply:
The turret can be destroyed while the component was scheduling a shot. Since qdeleting the comp could lead to issue in the autofire, i did this

Reply:
add a detailed comment for explanation